libs/s25main/world/MapGeometry.cpp

```diff
@@ -32,15 +34,17 @@ Position GetNeighbour(const Position& p, const Direction dir)
                 SE   0|1    1|1
                 SW  -1|1    0|1
     */
-    switch(dir.native_value())
+    switch(dir)
     {
         case Direction::WEST: return Position(p.x - 1, p.y);
         case Direction::NORTHWEST: return Position(p.x - !(p.y & 1), p.y - 1);
         case Direction::NORTHEAST: return Position(p.x + (p.y & 1), p.y - 1);
         case Direction::EAST: return Position(p.x + 1, p.y);
         case Direction::SOUTHEAST: return Position(p.x + (p.y & 1), p.y + 1);
-        default: RTTR_Assert(dir == Direction::SOUTHWEST); return Position(p.x - !(p.y & 1), p.y + 1);
+        case Direction::SOUTHWEST: return Position(p.x - !(p.y & 1), p.y + 1);
     }
+    RTTR_Assert(false);
```

Comment:
we should probably add the possibility to add a text to the asserts, so we can make it more readable whats happening?
like
```
RTTR_Assert(false) << "Unknown direction occured";
```
or something like that? (like the GTEST assertions?)

Reply:
That would make it much more heavy-weight. You can already do `RTTR_Assert(!"Unknown direction occured");` similar to `assert` on which this is based on.
And I think the usages are very rare. Usually it is a simple precondition that is checked which is expressive enough, if not a comment helps. And here it is above a `BOOST_UNREACHABLE_RETURN`, so that should be clear too.

Reply:
ok thats true in that case

Reply:
Just seen I already did that: RTTR_Assert_Msg which is a readable shortcut for the above workaround